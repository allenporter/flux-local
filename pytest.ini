[pytest]
log_level = DEBUG
asyncio_mode = auto
