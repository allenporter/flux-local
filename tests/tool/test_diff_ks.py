"""Tests for the flux-local `diff ks` command."""

import pytest

from syrupy.assertion import SnapshotAssertion

from . import run_command


@pytest.mark.parametrize(
    ("args", "env"),
    [
        (["apps", "--path", "tests/testdata/cluster"], {}),
        (["apps", "--path", "tests/testdata/cluster"], {"DIFF": "diff"}),
        (["apps", "--path", "tests/testdata/cluster", "-o", "yaml"], {}),
        (
            [
                "apps",
                "--path",
                "tests/testdata/cluster",
                "-o",
                "yaml",
                "--limit-bytes",
                "20000",
            ],
            {},
        ),
        (
            [
                "apps",
                "--path",
                "tests/testdata/cluster",
                "-o",
                "yaml",
                "--sources",
                "''",
            ],
            {},
        ),
    ],
    ids=["apps", "ks-external", "yaml", "yaml-limit", "yaml-empty-sources"],
)
async def test_diff_ks(
    args: list[str], env: dict[str, str], snapshot: SnapshotAssertion
) -> None:
    """Test test diff ks commands."""
    result = await run_command(["diff", "ks"] + args, env=env)
    assert result == snapshot
