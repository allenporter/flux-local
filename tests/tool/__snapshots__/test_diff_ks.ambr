# serializer version: 1
# name: test_diff_ks[apps]
  ''
# ---
# name: test_diff_ks[ks-external]
  ''
# ---
# name: test_diff_ks[yaml-empty-sources]
  '''
  ---
  - kustomization_path: tests/testdata/cluster/apps/prod
    kind: Kustomization
    namespace: flux-system
    name: apps
    diffs:
    - kustomization_path: tests/testdata/cluster/apps/prod
      kind: Namespace
      namespace: flux-system
      name: podinfo
      diff: |-
        --- tests/testdata/cluster/apps/prod Kustomization: flux-system/apps Namespace: flux-system/podinfo
  
        +++ tests/testdata/cluster/apps/prod Kustomization: flux-system/apps Namespace: flux-system/podinfo
  
        @@ -1,9 +0,0 @@
  
        ----
        -apiVersion: v1
        -kind: Namespace
        -metadata:
        -  labels:
        -    kustomize.toolkit.fluxcd.io/name: apps
        -    kustomize.toolkit.fluxcd.io/namespace: flux-system
        -  name: podinfo
        -
    - kustomization_path: tests/testdata/cluster/apps/prod
      kind: HelmRelease
      namespace: podinfo
      name: podinfo
      diff: |-
        --- tests/testdata/cluster/apps/prod Kustomization: flux-system/apps HelmRelease: podinfo/podinfo
  
        +++ tests/testdata/cluster/apps/prod Kustomization: flux-system/apps HelmRelease: podinfo/podinfo
  
        @@ -1,37 +0,0 @@
  
        ----
        -apiVersion: helm.toolkit.fluxcd.io/v2beta1
        -kind: HelmRelease
        -metadata:
        -  labels:
        -    kustomize.toolkit.fluxcd.io/name: apps
        -    kustomize.toolkit.fluxcd.io/namespace: flux-system
        -  name: podinfo
        -  namespace: podinfo
        -spec:
        -  chart:
        -    spec:
        -      chart: podinfo
        -      sourceRef:
        -        kind: HelmRepository
        -        name: podinfo
        -        namespace: flux-system
        -      version: 6.3.2
        -  install:
        -    remediation:
        -      retries: 3
        -  interval: 50m
        -  releaseName: podinfo
        -  values:
        -    ingress:
        -      className: nginx
        -      enabled: true
        -      hosts:
        -      - host: podinfo.production
        -        paths:
        -        - path: /
        -          pathType: ImplementationSpecific
        -    redis:
        -      enabled: true
        -      repository: public.ecr.aws/docker/library/redis
        -      tag: 7.0.6
        -
    - kustomization_path: tests/testdata/cluster/apps/prod
      kind: ConfigMap
      namespace: podinfo
      name: podinfo-config
      diff: |-
        --- tests/testdata/cluster/apps/prod Kustomization: flux-system/apps ConfigMap: podinfo/podinfo-config
  
        +++ tests/testdata/cluster/apps/prod Kustomization: flux-system/apps ConfigMap: podinfo/podinfo-config
  
        @@ -1,12 +0,0 @@
  
        ----
        -apiVersion: v1
        -data:
        -  foo: bar
        -kind: ConfigMap
        -metadata:
        -  labels:
        -    kustomize.toolkit.fluxcd.io/name: apps
        -    kustomize.toolkit.fluxcd.io/namespace: flux-system
        -  name: podinfo-config
        -  namespace: podinfo
        -
  
  
  '''
# ---
# name: test_diff_ks[yaml-limit]
  ''
# ---
# name: test_diff_ks[yaml]
  ''
# ---
